test/examples/to-fix/circuit/Datatypes.ju

```diff
@@ -25,6 +25,19 @@ sig patternMatch : Bar -> field
 let patternMatch (P x y) = x + y
 let patternMatch (Q x y z) = x + y + z
 
+-- Fails to typecheck:
+-- "Error on toHR: Name "AtomExp" :| [] not in scope
+-- ( probably a bug in the elaborator from frontend )"
+type AtomExp a = Atom a
+
+sig testAExp : AtomExp field
+let testAExp = Atom 1
+
+-- Fails to typecheck:
+-- "Error on toHR: Name "SExp" :| [] not in scope
+-- ( probably a bug in the elaborator from frontend )"
+type SExp a = Atom a | SPair (SExp a) (SExp a)
```

Comment:
This brings up an intersting test case we should have.

Namely in a working example can we break it with

```haskell
type atom-exp a = Atom a

type more-fun-exp = Atom Int.t
```

I think this is an error waiting to happen

Reply:
Good idea, I added that one.